src/post/Post.tsx

```diff
@@ -7,137 +8,134 @@ import {
   TagsType,
   UserIdentityType,
 } from "types";
-import {
-  faCertificate,
-  faCrown,
-  faEllipsisV,
-  faPlus,
-} from "@fortawesome/free-solid-svg-icons";
 
-import Badge from "./Badge";
-import Card from "common/Card";
 import Editor from "@bobaboard/boba-editor";
-import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import Footer from "./Footer";
+import Icon from "common/Icon";
 import type { IconProps } from "common/Icon";
+import PostPreamble from "./PostPreamble";
 import React from "react";
-import Reaction from "common/Reaction";
 import Tags from "tags/Tags";
 import TagsFactory from "tags/TagsFactory";
 import Theme from "theme/default";
-import UpdatesHeader from "./UpdatesHeader";
 import classnames from "classnames";
+import css from "styled-jsx/css";
 import debug from "debug";
+import { faEllipsisV } from "@fortawesome/free-solid-svg-icons";
 
 const log = debug("bobaui:post-log");
 
-export const modes = {
-  VIEW: "VIEW",
-  CREATE: "CREATE",
-};
-
 const COLLAPSED_HEIGHT = 150;
 
 const MemoizedHeader = React.memo(Header);
 const MemoizedEditor = React.memo(Editor);
+const MemoizedFooter = React.memo(Footer);
 
-const PostFooter: React.FC<PostProps> = (props) => {
-  return (
-    <div
-      className={classnames("footer", {
-        "with-reactions": !!props.reactions?.length,
-        muted: props.muted,
-        hidden: props.hideFooter,
-      })}
-    >
-      <div className="notes">
-        <Footer
-          onContribution={props.onNewContribution}
-          onComment={props.onNewComment}
-          totalContributions={props.totalContributions}
-          directContributions={props.directContributions}
-          totalComments={props.totalComments}
-          newContributions={props.newContributions}
-          newComments={props.newComments}
-          notesLink={props.notesLink}
-          allowsContribution={props.allowsContribution}
-          allowsComment={props.allowsComment}
-        />
-      </div>
-      {!!props.reactable && (
-        <div className="reactions">
-          {props.reactions?.map((reaction) => (
-            <div className="reaction" key={reaction.image}>
-              <Reaction image={reaction.image} count={reaction.count} />
-            </div>
-          ))}
-          <div className="add-reaction">
-            <FontAwesomeIcon icon={faPlus} />
-          </div>
-        </div>
+const { styles: cardStyles, className: cardClassName } = css.resolve`
+  footer.with-tags {
+    border-top: 1px dotted #d2d2d2;
+  }
+  footer {
+    display: flex;
+    flex-direction: column;
+    gap: 8px;
+  }
+  header:not(.with-board) {
+    padding-bottom: 4px;
+    border-bottom: 1px dotted #d2d2d2;
+  }
+  .content {
+    --text-color: ${Theme.POST_TEXT_COLOR};
+  }
+`;
+
+const { styles: headerStyles, className: headerClassName } = css.resolve`
+  .metadata {
+    max-width: calc(100% - 20px);
+  }
+  button {
```

Comment:
Very fair. OTOH I don't think this is always true, for example "header/footer" in the cards I feel is clear enough (because usually you always have one). But I very much agree with the general principle.

It would be nice for us to come up with guidelines for this, but maybe let's give it a little bit to see how we feel about doing stuff this way in the future.

FWIW, I could put this before or after, I haven't actually figured out what the right position is.